src/pages/VaccineModulePage.js

```diff
@@ -0,0 +1,281 @@
+        {hasBreaches && (
+          <View style={[fridgeInfoSectionStyle, { marginRight: 10 }]}>
+            <Text style={greyTextStyleSmall}>Breaches:</Text>
+            <Text style={greyTextStyleLarge}>{numberOfBreaches}</Text>
+          </View>
+        )}
+        <Text style={greyTextStyleSmall}>Exposure:</Text>
+        {this.renderTemperature(20, minTemperature)}
+        <Text style={greyTextStyleSmall}>to</Text>
+        {this.renderTemperature(20, maxTemperature)}
+      </View>
+    );
+  };
+
+  /* Render 'Breach: {num} Exposure: {fromTemp} to {toTemp} */
+  renderFridgeStock = fridge => {
+    const { navigateTo, database } = this.props;
+    const fridgeStock = fridge.getTotalStock(database);
+    const hasStock = fridgeStock > 0;
+
+    const onPress = () =>
+      navigateTo('manageVaccineItems', 'Manage Vaccine Items', {
+        initialLocation: fridge,
+      });
+
+    const { fridgeInfoSectionStyle, greyTextStyleLarge, greyTextStyleSmall } = localStyles;
+    const fontStyleLarge = { ...greyTextStyleLarge, color: SUSSOL_ORANGE };
+    const fontStyleSmall = { ...greyTextStyleSmall, color: SUSSOL_ORANGE };
+
+    if (!hasStock) return <Text style={fontStyleSmall}>Empty Fridge</Text>;
+    return (
+      <TouchableOpacity style={fridgeInfoSectionStyle} onPress={onPress}>
+        <Text style={fontStyleSmall}>Total Stock:</Text>
+        <Text style={fontStyleLarge}>{fridgeStock}</Text>
+        {this.renderIcon('angle-double-right', CHEVRON_ICON_STYLE)}
+      </TouchableOpacity>
+    );
+  };
+
+  /* Fridge and all of it's components */
+  renderFridge = fridge => {
+    const { database } = this.props;
+    const currentTemperature = fridge.getCurrentTemperature(database);
+    const isCriticalTemperature = fridge.isCriticalTemperature(database);
+
+    const { selectedFridgeId } = this.state;
+    const isFridgeSelected = fridge.code === selectedFridgeId;
+
+    const { sectionStyle, fridgeInfoSectionStyle } = localStyles;
+    return (
+      <View key={fridge.code}>
+        <View style={[sectionStyle, { flexDirection: 'column', alignItems: 'stretch' }]}>
+          <View style={fridgeInfoSectionStyle}>
+            {this.renderFridgeName(fridge, isFridgeSelected)}
+            {currentTemperature !== null ? currentTemperature : null}
+
+            {isCriticalTemperature ? this.renderIcon('warning', BREACH_ICON_STYLE) : null}
+            <View style={[fridgeInfoSectionStyle, { justifyContent: 'flex-end', flexGrow: 1 }]}>
+              {this.renderFridgeExtraInfo(fridge)}
+              {this.renderFridgeStock(fridge)}
+            </View>
+          </View>
+          {isFridgeSelected && (
+            <View style={{ height: 250, alignSelf: 'stretch' }}>
+              <VaccineChart {...fridge.getTemperaturePoints()} />
+            </View>
+          )}
+        </View>
+      </View>
+    );
+  };
+
+  render() {
+    const { fridges } = this.state;
```

Comment:
`fridges` is not guaranteed to have been set yet on the first render. I think `if (!fridges) return null` is a good idea

Reply:
i think this should be fine as this.hasFridges will be null still.